### PR TITLE
fix formatting issues with large UIDs

### PR DESCRIPTION
**Pull request by grondo**
As identified by @wkharold in #4486, uids are treated as signed 32bit integers in several places, including packing into JSON objects like message payloads and eventlogs, as well as when the shell service string is rendered.

I ran `make check` in a docker container as a user with a large UID and this PR is the result of fixing test failures.

I tried to update all places where a userid is packed into a JSON object to pack as `json_int_t` (with a cast) (`json_pack ("I")`), while still unpacking as `i` (since the destination is still 32 bits). @garlick and I could not think of why this wouldn't work.

It would be nice to run one of the CI builds with a VLU (very large userid), but I wasn't quite sure how to get that done quickly and wanted to get this fix in first.

### CODE REVIEW

File: src/modules/job-manager/urgency.c [hunk 1]
```diff
@@ -97,7 +97,7 @@ void urgency_handle_request (flux_t *h,
     orig_urgency = job->urgency;
     if (event_job_post_pack (ctx->event, job,
                              "urgency", 0,
-                             "{ s:i s:i }",
+                             "{ s:I s:i }",
                              "userid", cred.userid,
                              "urgency", urgency) < 0)
```
Comment (garlick) [inline src/modules/job-manager/urgency.c:102]: missed the typecast on this one

**Review by garlick — APPROVED** [review]
LGTM - just one nit

File: src/modules/job-manager/urgency.c (hunk 1 above)
Comment (grondo) [reply src/modules/job-manager/urgency.c]: Nice catch! Thanks, I've fixed that one and pushed the result.

**Comment by grondo** [comment]
Setting MWP.

**Pull request merged**